### PR TITLE
[fix] Make NotifyReleaseStages apply to sessions

### PR DESCRIPTION
**Pull request by kinbiko**
We were previously still sending sessions if the release stage was outside a defined set of acceptable notify release stages. This PR fixes that.

### CODE REVIEW

File: sessions/publisher.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ func (p *publisher) publish(sessions []*Session) error {
 		// log every minute
 		return nil
 	}
+	nrs, rs := p.config.NotifyReleaseStages, p.config.ReleaseStage
+	if nrs != nil && !contains(nrs, rs) {
```
Comment (CodeHex) [inline sessions/publisher.go]: We should add a test and check that delivers when release stage is "" and notify release stages is nil

Comment (kinbiko) [reply sessions/publisher.go]: Added a test for release stage being `""`. Most of the integration tests already verify that `NotifyReleaseStages` being nil still results in the session payload being published.
